rules/library.bzl

```diff
@@ -142,8 +142,9 @@ def _write_umbrella_header(
         module_name = name
 
     content = ""
-
+    extern_keyword = "extern"
     if generate_default_umbrella_header:
+        extern_keyword = "FOUNDATION_EXPORT"
         content += """\
 #ifdef __OBJC__
```

Comment:
Could we always keep the else clause so that we always use FOUNDATION_EXPORT. It will be something like
```
if generate_default_umbrella_header:
        content += """\
#ifdef __OBJC__
#    import <Foundation/Foundation.h>
#    if __has_include(<UIKit/UIKit.h>)
#        import <UIKit/UIKit.h>
#    endif
#endif
"""

content += """\
#ifndef __OBJC__
#    ifndef FOUNDATION_EXPORT
#        if defined(__cplusplus)
#            define FOUNDATION_EXPORT extern "C"
#        else
#            define FOUNDATION_EXPORT extern
#        endif
#    endif
#endif
```

Reply:
this will end having ` error: unknown type name 'FOUNDATION_EXPORT'` error if we also replace `extern` with `FOUNDATION_EXPORT`.
this is because when generate_default_umbrella_header is off and the code has `ifdef __OBJC__` is true, you won't get the FOUNDATION_EXPORT macro defined.

If we add just 
```
#    ifndef FOUNDATION_EXPORT
#        if defined(__cplusplus)
#            define FOUNDATION_EXPORT extern "C"
#        else
#            define FOUNDATION_EXPORT extern
#        endif
#    endif
```
we end up having the `warning: ambiguous expansion of macro 'FOUNDATION_EXPORT' error.` which is sth i don't want

Reply:
That's weird. Could it be that we need to always `define FOUNDATION_EXPORT extern` when `ifdef __OBJC__` is true (even if defined(__cplusplus) is true)? Could you try
```
if generate_default_umbrella_header:
    content += """\
#ifdef __OBJC__
#    import <Foundation/Foundation.h>
#    if __has_include(<UIKit/UIKit.h>)
#        import <UIKit/UIKit.h>
#    endif
#else
#    ifndef FOUNDATION_EXPORT
#        if defined(__cplusplus)
#            define FOUNDATION_EXPORT extern "C"
#        else
#            define FOUNDATION_EXPORT extern
#        endif
#    endif
#endif
"""
else:
    content += """\
#ifdef __OBJC__
#    define FOUNDATION_EXPORT extern
#else
#    ifndef FOUNDATION_EXPORT
#        if defined(__cplusplus)
#            define FOUNDATION_EXPORT extern "C"
#        else
#            define FOUNDATION_EXPORT extern
#        endif
#    endif
#endif
"""
```

Reply:
exactly, might as well be very explicit about the two code path this is going with